pkg/providers/openai/chat.go

```diff
@@ -147,8 +146,56 @@ func (c *Client) doChatRequest(ctx context.Context, payload *ChatRequest) (*sche
 		return nil, errs.ErrProviderUnavailable
 	}
 
+	// Read the response body into a byte slice
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		c.telemetry.Logger.Error("failed to read openai chat response", zap.Error(err))
+		return nil, err
+	}
+
+	// Parse the response JSON
+	var responseJSON map[string]interface{}
+
+	err = json.Unmarshal(bodyBytes, &responseJSON)
+	if err != nil {
+		c.telemetry.Logger.Error("failed to parse openai chat response", zap.Error(err))
+		return nil, err
+	}
+
 	// Parse response
 	var response schemas.UnifiedChatResponse
 
-	return &response, json.NewDecoder(resp.Body).Decode(&response)
+	var responsePayload schemas.ProviderResponse
+
+	var tokenCount schemas.TokenCount
+
+	message := responseJSON["choices"].([]interface{})[0].(map[string]interface{})["message"].(map[string]interface{})
```

Comment:
@mkrueger12 looking at this code I feel like it's worth defining OpenAI Chat Response schema, map the response into it and then remap into the unified chat response schema

Reply:
@roma-glushko  I can do that. What advantage does it provide?

Reply:
@mkrueger12 That would make the implementation more readable at very least and it would give an understanding what the actual provider response might look like, what information it contains in a declarative way 👀 

In this specific case, expression like `responseJSON["choices"].([]interface{})[0].(map[string]interface{})` means that you are sure that the response has the `choices` key where you also are sure there is at least one item e.g. `[0]`. But is this always a case? Because if not I think this code will fail with an exception trying to access that info but in fact getting `nil`.

If you had a validation via the defined OpenAI chat response, you would ensure that the response has all needed fields which would simply the code here. 

Something like that.